oneview_redfish_toolkit/api/storage.py

```diff
@@ -36,84 +38,121 @@
-        self.redfish["@odata.context"] = \
-            "/redfish/v1/$metadata#Storage.Storage"
-        self.redfish["@odata.type"] = self.get_odata_type()
-        self.redfish["StorageControllers@odata.count"] = 1
-        self.redfish["Id"] = "1"
-        self.redfish["Name"] = "Storage Controller"
-        self.redfish["Status"] = collections.OrderedDict()
+        attrs["StorageControllers@odata.count"] = 1
+        attrs["Id"] = "1"
+        attrs["Name"] = "Storage Controller"
+        attrs["Status"] = collections.OrderedDict()
         ok_struct = status_mapping.STATUS_MAP.get("OK")
-        self.redfish["Status"]["State"] = ok_struct["State"]
-        self.redfish["Status"]["Health"] = ok_struct["Health"]
-        self.redfish["StorageControllers"] = list()
+        attrs["Status"]["State"] = ok_struct["State"]
+        attrs["Status"]["Health"] = ok_struct["Health"]
+        attrs["StorageControllers"] = list()
 
         # adapter storage capabilities (if any)
         for adapter in server_hardware_type['adapters']:
             if adapter['storageCapabilities']:
                 new_capability = collections.OrderedDict()
                 new_capability["SupportedDeviceProtocols"] = sorted(
-                    self.map_supported_device_protos(drive_technologies))
-                self.redfish["StorageControllers"].append(new_capability)
+                    Storage._map_supported_device_protos(drive_technologies))
+                attrs["StorageControllers"].append(new_capability)
 
         # internal storage capabilities
         storage_controllers = collections.OrderedDict()
         storage_controllers["Manufacturer"] = "HPE"
         storage_controllers["SupportedDeviceProtocols"] = \
-            sorted(self.map_supported_device_protos(drive_technologies))
-        self.redfish["StorageControllers"].append(storage_controllers)
+            sorted(Storage._map_supported_device_protos(drive_technologies))
+        attrs["StorageControllers"].append(storage_controllers)
 
         count_drives_by_jbod = \
             [int(item["numPhysicalDrives"]) for item in sas_logical_jbods]
 
         count_drives = reduce(operator.add, count_drives_by_jbod, 0)
 
-        self.redfish["Drives@odata.count"] = count_drives
-        self.redfish["Drives"] = list()
+        attrs["Drives@odata.count"] = count_drives
+        attrs["Drives"] = list()
         for i in range(count_drives):
             drive_id = str(i + 1)
-            self.redfish["Drives"].append({
-                "@odata.id": self.redfish["@odata.id"] + "/Drives/" + drive_id
+            attrs["Drives"].append({
+                "@odata.id": attrs["@odata.id"] + "/Drives/" + drive_id
             })
 
-        self._validate()
+        return Storage(attrs)
 
-    def map_supported_device_protos(self, drive_technologies):
-        supported_device_protocols = set()
+    @staticmethod
+    def build_for_resource_blcok(drive):
```

Comment:
Fix the typo in the method name "build_for_resource_blcok"

Reply:
haha....thanks!